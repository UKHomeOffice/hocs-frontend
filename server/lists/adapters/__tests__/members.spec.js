const membersAdapter = require('../members');

describe('Members Adapter', () => {

    it('should transform member data', async () => {
        const mockData = {
            members: [
                { group: 'A', label: 'Member 1', value: 'MEMBER_1' },
                { group: 'A', label: 'Member 2', value: 'MEMBER_2' },
                { group: 'B', label: 'Member 3', value: 'MEMBER_3' },
                { group: 'C', label: 'Member 4', value: 'MEMBER_4' }
            ]
        };

        const results = await membersAdapter(mockData);

        expect(results).toBeDefined();
        expect(results).toMatchSnapshot();
    });
});