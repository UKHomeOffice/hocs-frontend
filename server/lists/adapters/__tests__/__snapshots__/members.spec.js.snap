// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members Adapter should transform member data 1`] = `
Array [
  Object {
    "label": "A",
    "options": Array [
      Object {
        "label": "Member 1",
        "value": "MEMBER_1",
      },
      Object {
        "label": "Member 2",
        "value": "MEMBER_2",
      },
    ],
  },
  Object {
    "label": "B",
    "options": Array [
      Object {
        "label": "Member 3",
        "value": "MEMBER_3",
      },
    ],
  },
  Object {
    "label": "C",
    "options": Array [
      Object {
        "label": "Member 4",
        "value": "MEMBER_4",
      },
    ],
  },
]
`;
